libs/experimental/langchain_experimental/data_anonymizer/deanonymizer_mapping.py

```diff
@@ -17,5 +20,75 @@ def data(self) -> MappingDataType:
         return {k: dict(v) for k, v in self.mapping.items()}
 
     def update(self, new_mapping: MappingDataType) -> None:
+        """Update the deanonymizer mapping with new values
+        Duplicated values will not be added
+        """
+        new_values_seen = set()
+
         for entity_type, values in new_mapping.items():
-            self.mapping[entity_type].update(values)
+            for k, v in values.items():
+                # Make sure it is not a duplicate value
+                if (
+                    v not in self.mapping[entity_type].values()
+                    and v not in new_values_seen
+                ):
+                    self.mapping[entity_type][k] = v
+                    new_values_seen.update({v})
+
+
+def create_anonymizer_mapping(
+    original_text: str,
+    analyzer_results: List[RecognizerResult],
+    anonymizer_results: EngineResult,
+    reversed: bool = False,
```

Comment:
There is a python function named `reversed` - I'm not sure if it's good idea to overwrite it. Maybe `is_reversed` ?